src/devtools/views/Components/OwnersStack.js

```diff
@@ -1,43 +1,85 @@
 // @flow
-
-import React, { useCallback, useContext } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+  createRef,
+  forwardRef,
+} from 'react';
+import throttle from 'lodash.throttle';
+import classNames from 'classnames';
 import Button from '../Button';
 import ButtonIcon from '../ButtonIcon';
 import { TreeContext } from './TreeContext';
 import { StoreContext } from '../context';
+import { getElementDimensions } from '../../../utils';
 
 import type { Element } from './types';
 
 import styles from './OwnersStack.css';
 
-export default function OwnerStack() {
-  const { ownerStack, resetOwnerStack } = useContext(TreeContext);
+type ElementsDropdownProps = {
+  selectedElementIndex: number | null,
+  children: Array<any>,
+};
+function ElementsDropdown({
+  selectedElementIndex,
+  children,
+}: ElementsDropdownProps) {
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const handleClick = useCallback(() => {
+    setIsDropdownVisible(!isDropdownVisible);
+  }, [isDropdownVisible, setIsDropdownVisible]);
 
-  const elements = ownerStack.map((id, index) => (
-    <ElementView key={id} id={id} index={index} />
-  ));
+  useEffect(() => {
+    setIsDropdownVisible(false);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedElementIndex]);
 
   return (
-    <div className={styles.OwnerStack}>
+    <div className={styles.ElementsDropdown}>
       <Button
-        className={styles.IconButton}
-        onClick={resetOwnerStack}
-        title="Back to tree view"
+        className={classNames(styles.IconButton, {
+          [styles.DropdownButtonActive]: isDropdownVisible,
+        })}
+        onClick={handleClick}
+        title="Open elements dropdown"
       >
-        <ButtonIcon type="close" />
+        <ButtonIcon type="colon" />
       </Button>
-      <div className={styles.VRule} />
-      {elements}
+      {isDropdownVisible && <div className={styles.Dropdown}>{children}</div>}
     </div>
   );
 }
 
-type Props = {
+type ElementsBarProps = {
+  elements: Array<any>,
+  showSelectedOnly: boolean,
+};
+const ElementsBar = forwardRef(
+  (
+    { elements, showSelectedOnly }: ElementsBarProps,
+    ref: Object
+  ) => {
+    return (
+      <div
+        className={classNames(styles.ElementsBar, {
+          [styles.ElementsBarSelectedOnly]: showSelectedOnly,
+        })}
+        ref={ref}
```

Comment:
We are adding this ref in order to track the width of the ElementsBar and its children.